pkg/edge/ICoreWebView2NavigationStartingEventArgs.go

```diff
@@ -0,0 +1,21 @@
+package edge
+
+type _ICoreWebView2NavigationStartingEventArgsVtbl struct {
```

Comment:
There are two problems here:

1. Like `ICoreWebView2NavigationCompletedEventArgs`, there is a corresponding extended event args object for `ICoreWebView2NavigationStartingEventArgs` called, predictably, `ICoreWebView2NavigationStartingEventArgs2`. As I outlined with `GetHttpStatusCode`, it is not safe to include these directly without calling `QueryInterface` first. The GUID for `ICoreWebView2NavigationStartingEventArgs2` is `9086BE93-91AA-472D-A7E0-579F2BA006AD`.
2. This Vtable is out of order, which will lead to the wrong functions being called.

The Vtable order for `ICoreWebView2NavigationStartingEventArgs` is:

```c++
[propget] HRESULT Uri([out, retval] LPWSTR* uri);
[propget] HRESULT IsUserInitiated([out, retval] BOOL* isUserInitiated);
[propget] HRESULT IsRedirected([out, retval] BOOL* isRedirected);
[propget] HRESULT RequestHeaders([out, retval] ICoreWebView2HttpRequestHeaders** requestHeaders);
[propget] HRESULT Cancel([out, retval] BOOL* cancel);
[propput] HRESULT Cancel([in] BOOL cancel);
[propget] HRESULT NavigationId([out, retval] UINT64* navigationId);
```

(Note that your names are fine. The `propget`/`propput` is syntax sugar in COM.)

And `ICoreWebView2NavigationStartingEventArgs2`:

```c++
// (... everything from the original ICoreWebView2NavigationStartingEventArgs)
[propget] HRESULT AdditionalAllowedFrameAncestors([out, retval] LPWSTR* value);
[propput] HRESULT AdditionalAllowedFrameAncestors([in] LPCWSTR value);
```

(Note that there is a put from `AdditionalAllowedFrameAncestors` _as well as_ the get you already have.)

If you are wondering where this information comes from, it comes from the WebView2 NuGet package. You can download that here:

https://www.nuget.org/packages/Microsoft.Web.WebView2

Just click "Download Package". The corresponding `nupkg` file is a ZIP archive, and once you unzip it you will see `WebView2.idl` which contains the full COM IDL for all WebView2-related classes.

MSDN alone is not good enough, as MSDN displays the functions in a different order from their VTable ordering. We need these functions to be in VTable order for our memory map to align.